File: src/Find the Peak Element in a 2D Matrix/Code/solution.py
class Solution:
    def findPeakGrid(self, mat):
        n = len(mat)
        m = len(mat[0])

        for i in range(n):
            for j in range(m):
                curr = mat[i][j]

                up = mat[i - 1][j] if i > 0 else float('-inf')
                down = mat[i + 1][j] if i < n - 1 else float('-inf')
                left = mat[i][j - 1] if j > 0 else float('-inf')
                right = mat[i][j + 1] if j < m - 1 else float('-inf')

                if curr >= up and curr >= down and \
                   curr >= left and curr >= right:
                    return [i, j]

        return [-1, -1]

File: src/Find the Peak Element in a 2D Matrix/Code/solution.js
class Solution {
  findPeakGrid(mat) {
    let n = mat.length;
    let m = mat[0].length;

    for (let i = 0; i < n; i++) {
      for (let j = 0; j < m; j++) {
        let curr = mat[i][j];

        let up = i > 0 ? mat[i - 1][j] : -Infinity;
        let down = i < n - 1 ? mat[i + 1][j] : -Infinity;
        let left = j > 0 ? mat[i][j - 1] : -Infinity;
        let right = j < m - 1 ? mat[i][j + 1] : -Infinity;

        if (curr >= up && curr >= down && curr >= left && curr >= right) {
          return [i, j];
        }
      }
    }
    return [-1, -1];
  }
}
